### PR TITLE
Issues #17, #16

### PR DESCRIPTION
**Pull request by ttousai**
Support --installation option. Currently supports kops, hyperkube, kubeadm and the default installation. Will need some fleshing out for the bin/conf paths (#17).

Also added a --verbose flag to print verbose messages for debugging (#16).

### CODE REVIEW

File: cmd/common.go [hunk 1]
```diff
@@ -129,34 +269,27 @@ func cleanIDs(list string) []string {
 // Any check failing here is a show stopper.
```
Comment (lizrice) [inline cmd/common.go]: Should have spotted this before but this comment isn't true

File: cmd/common.go [hunk 2]
```diff
@@ -256,36 +389,31 @@ func verifyBin(binPath []string) []string {
...
-func verifyKubeVersion(binPath []string) []string {
+func verifyKubeVersion(b string) []string {
 	// These executables might not be on the user's path.
 	// TODO! Check the version number using kubectl, which is more likely to be on the path.
 	var w []string
 
-	for _, b := range binPath {
-		_, err := exec.LookPath(b)
-		if err != nil {
-			w = append(w, fmt.Sprintf("%s: command not found on path - version check skipped\n", b))
```
Comment (lizrice) [inline cmd/common.go]: Let's keep this warning as part of the output if the command isn't found on the path - this is useful information to have even if --verbose is not on.

File: cmd/common.go [hunk 3]
```diff
@@ -256,36 +389,31 @@ func verifyBin(binPath []string) []string {
...
+	cmd := exec.Command(b, "--version")
+	out, err := cmd.Output()
+	errmsgs += handleError(
+		err,
+		fmt.Sprintf("verifyKubeVersion: failed\nCommand:%s", cmd.Args),
+	)
 
-		matched := strings.Contains(string(out), kubeVersion)
-		if !matched {
-			w = append(w, fmt.Sprintf(
```
Comment (lizrice) [inline cmd/common.go]: Any reason not to leave in the "expected" and "got" in the warning output? It seems potentially useful information

**Comment by ttousai** [comment]
Nice! This is pretty detailed! I will review and update.

File: cmd/common.go (hunk 2 above)
Comment (ttousai) [inline cmd/common.go]: I think I will have to spend sometime looking at error handling as a whole. For now I will restore this message but only on asserting the verbose flag.

File: check/check.go [hunk 4]
```diff
@@ -88,38 +96,50 @@ func (c *Check) Run() {
...
 	// Start command pipeline
 	i = 0
 	for i < n {
 		err := cs[i].Start()
-		if err != nil {
-			fmt.Fprintf(os.Stderr, "%s: %s\n", cs[i].Args, err)
-		}
+		errmsgs += handleError(
+			err,
+			fmt.Sprintf("check.Run: Audit %s failed\nfailing command: %s",
```
Comment (ttousai) [inline check/check.go]: I try to capture 2 pieces of info with that error message:
1. The entire audit pipeline that failed.
2. Which specific command in that pipeline failed.

So something like:
`
Failed to run: commandline
Failing command: actual command
`

File: cmd/common.go (hunk 1 above)
Comment (ttousai) [inline cmd/common.go]: Missed it too!

File: cmd/common.go (hunk 3 above)
Comment (ttousai) [inline cmd/common.go]: I run into issues where the error message looks like;
```
expected: 1.6, got: 
```
This usually happens when the kubernetes binary I'm using for version check is not running. So I tried to solve but removing this error message and replacing it with something more generic.

**Pull request merged**